libraries/chain/database.cpp

```diff
@@ -2252,47 +2262,83 @@ namespace golos { namespace chain {
             try {
                 uint128_t total_weight(c.total_vote_weight);
                 //edump( (total_weight)(max_rewards) );
+                
                 share_type unclaimed_rewards = max_rewards;
 
+                uint128_t auction_window_reward = uint128_t(1) * max_rewards.value * c.auction_window_weight / total_weight;
+                auto votes_after_auction_window_weight = total_weight - c.votes_in_auction_window_weight - c.auction_window_weight;
+
+                auto auw_time = c.created + c.auction_window_size;
+
                 if (c.total_vote_weight > 0 && c.allow_curation_rewards) {
+                    // If auction window reward is chosen to go to curators, we need to
+                    // separated votes for 2 sets: 
+                    // c.created     auction_window               cashout
+                    // |_______________|____________________________|
+                    //     1                      2 
+                    // 1st set contains votes with last_update from [c.created, c.created + auction_window_size)
+                    // 2st set contains votes with last_update from [c.created + auction_window_size, cashout]
+
+                    // Also we need to distibute auction_window reward to voters from second set.
+                    // So we need to calculate parts of reward for 3 sets:
+                    //       - votes_in_auction_window_reward
+                    //       - votes_after_auction_window_reward
+                    //       - auction_window_reward 
+
                     const auto &cvidx = get_index<comment_vote_index>().indices().get<by_comment_weight_voter>();
                     auto itr = cvidx.lower_bound(c.id);
+
+                    // memorize the top weight account after auction window
+                    account_id_type heaviest_vote_after_auw_account;
+                    bool heaviest_vote_after_auw_founded = false;
+
                     while (itr != cvidx.end() && itr->comment == c.id) {
                         uint128_t weight(itr->weight);
-                        auto claim = ((max_rewards.value * weight) /
-                                      total_weight).to_uint64();
+                        uint64_t claim = ((max_rewards.value * weight) / total_weight).to_uint64();
+                        // to_curators case
+                        if (c.auction_window_reward_destination == protocol::to_curators && itr->last_update >= auw_time) {
+                            claim += ((auction_window_reward * weight) / votes_after_auction_window_weight).to_uint64();
+                            if (c.auction_window_reward_destination == protocol::to_curators && !heaviest_vote_after_auw_founded) {
```

Comment:
Duplicate checking on `c.auction_window_reward_destination == protocol::to_curators`
Remove it.